python/cog/director/redis.py

```diff
@@ -103,3 +126,32 @@ def rotate(self) -> None:
         self._current_consumer_index = (
             self._current_consumer_index + 1
         ) % consumer_count
+
+
+class RedisMessageClaimer:
+    def __init__(self, claim_message: Callable, claim_interval: int = 2):
+        self._thread = None
+        self._should_exit = threading.Event()
+        self._claim_message = claim_message
+        self._claim_interval = claim_interval
+
+    def start(self) -> None:
+        self._thread = threading.Thread(target=self._run)
+        self._thread.start()
+
+    def stop(self) -> None:
+        self._should_exit.set()
+        if self._thread is not None:
+            self._thread.join()
+
+    def _run(self) -> None:
+        last_claimed_at = time.perf_counter()
+
+        while not self._should_exit.is_set():
+            now = time.perf_counter()
+            if now >= last_claimed_at + self._claim_interval:
+                self._claim_message()
+                last_claimed_at = now
+
+            # Only sleep for a short time to respect _should_exit
+            time.sleep(0.01)
```

Comment:
This is so we don't block for 2 seconds when acking, right? Do we need to sleep at all if this is running on a thread?